iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -546,22 +544,14 @@ void PortManager::destroySubscriberPort(SubscriberPortType::MemberType_t* const
     LogDebug() << "Destroyed subscriber port";
 }
 
-runtime::IpcMessage PortManager::findService(const capro::IdString_t& service,
-                                             const capro::IdString_t& instance) noexcept
+runtime::IpcMessage PortManager::findService(const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& service,
+                                             const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& instance) noexcept
 {
-    // send find to all interfaces
-    capro::CaproMessage caproMessage(capro::CaproMessageType::FIND, {service, instance, roudi::Wildcard});
-
-    for (auto interfacePortData : m_portPool->getInterfacePortDataList())
-    {
-        iox::popo::InterfacePort interfacePort(interfacePortData);
-        interfacePort.dispatchCaProMessage(caproMessage);
-    }
-
```

Comment:
I also do not understand this, but I suspect it has something to do with the FIND message being changed and interface ports not being used properly (in the sense that they will not answer the request anyway).

But that is just a guess and requires clarification. @FerdinandSpitzschnueffler

Reply:
After changing the `findService` parameters, this code had to be adapted since the `CaproMessage` c'tor requires a `ServiceDescription`. I discussed several ideas with @elfenpiff, but then we came to the conclusion that the `CaproMessage` itself could need some refactoring and that we can send the `find` here but there will be no answer anyway as @MatthiasKillat stated.